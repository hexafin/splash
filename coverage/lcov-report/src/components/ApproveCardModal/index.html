<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/ApproveCardModal</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/ApproveCardModal
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.42% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">27.78% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">22.22% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.42% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>14/33</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="ApproveCardModal.js"><a href="ApproveCardModal.js.html">ApproveCardModal.js</a></td>
	<td data-value="42.42" class="pic low"><div class="chart"><div class="cover-fill" style="width: 42%;"></div><div class="cover-empty" style="width:58%;"></div></div></td>
	<td data-value="42.42" class="pct low">42.42%</td>
	<td data-value="33" class="abs low">14/33</td>
	<td data-value="27.78" class="pct low">27.78%</td>
	<td data-value="18" class="abs low">5/18</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="18" class="abs low">4/18</td>
	<td data-value="42.42" class="pct low">42.42%</td>
	<td data-value="33" class="abs low">14/33</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 02 2019 23:21:01 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
