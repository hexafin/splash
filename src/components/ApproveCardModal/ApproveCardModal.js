import React, { Component } from "react";
import { Animated, Easing, View, Text, TouchableOpacity, TouchableWithoutFeedback, StyleSheet, Image } from "react-native"
import { colors } from "../../lib/colors"
import LoadingCircle from "../universal/LoadingCircle"
import LetterCircle from "../universal/LetterCircle"
import Checkmark from "../universal/Checkmark"
import TouchID from "react-native-touch-id"
import Button from "../universal/Button"

class ApproveCardModal extends Component {

	constructor(props) {
		super(props)
		this.state = {
			success: false,
			wasLoading: false,
		}
	}

	componentWillMount() {
		this.backgroundOpacity = new Animated.Value(0)
	}

	componentDidMount() {
		Animated.sequence([
		Animated.delay(300),
		Animated.timing(this.backgroundOpacity, {
 			toValue: 1,
 			easing: Easing.linear(),
 			duration: 200
 		}),
	 ]).start();
	}

	componentWillReceiveProps(nextProps) { 
		if(nextProps.loading == true) {
			this.setState(prevState => {
				return {
					...prevState,
					wasLoading: true
				}
			})			
		}

		if(nextProps.loading == false && this.state.wasLoading == true) {
			this.setState(prevState => {
				return {
					...prevState,
					wasLoading: false,
					success: true,
				}
			})			
		}
	}

	render() {
		const {
			transactionId,
			relativeAmount,
			domain,
			relativeCurrency,
	    	exchangeRate
		} = this.props.navigation.state.params
		const transaction = {
			transactionId,
			relativeAmount,
			domain,
			relativeCurrency
		}

		const approve = transaction => {
			TouchID.authenticate("Approve Card")
				.then(success => {
					if (success) {
						this.props.ApproveTransaction(transaction)
					}
				})
				.catch(error => {
					console.log("TouchID Error:", error)
				})
		}

		const dismiss = () => {
			Animated.timing(this.backgroundOpacity, {
				toValue: 0,
				duration: 200,
				easing: Easing.linear(),
			}).start(({finished}) => {
				if (finished) {
					this.props.navigation.goBack()
					this.props.DismissTransaction()
				}
			})
		}

	  const letter = domain[0].toUpperCase()
	  const rate = parseFloat(exchangeRate).toFixed(2)
	  const btcAmount = (1.0*relativeAmount/parseFloat(exchangeRate)).toFixed(5)
		const domainCapitalized = domain[0].toUpperCase() + domain.slice(1)

		return (
			<TouchableWithoutFeedback onPress={() => dismiss()}>
			<Animated.View style={[styles.container, {backgroundColor: this.backgroundOpacity.interpolate({
																																										        inputRange: [0, 1],
																																										        outputRange: ['rgba(0, 0, 0, 0)', 'rgba(0, 0, 0, 0.2)']
																																										    })}]}>
				<View style={{ flexDirection: "row" }}>
					<TouchableWithoutFeedback onPress={() => {}}>
					<View style={styles.popup}>
						{ !this.props.error &&
								<View style={styles.content}>
	                <View style={styles.header}>
	                  <Text style={styles.title}>Transaction request</Text>
	                  <TouchableOpacity onPress={() => dismiss()}>
	                    <Image style={styles.closeButton} source={require('../../assets/icons/Xbutton.png')}/>
	                  </TouchableOpacity>
	                </View>
	                <View style={styles.information}>
	                  <Text style={styles.exchangeText}>1 BTC = ${rate} {relativeCurrency}</Text>
	                  <View style={styles.amountBox}>
	                    <Text style={{fontSize: 28, color: colors.white, fontWeight: '600'}}>{relativeCurrency} ${relativeAmount}</Text>
	                    <Text style={{fontSize: 18, opacity: 0.77, color: colors.white, fontWeight: '600'}}>{btcAmount} BTC</Text>
	                  </View>
	                  <View style={styles.domainInfo}>
	                    <Text style={{paddingRight: 10, fontSize: 12, fontWeight: '700', color: colors.nearBlack}}>on</Text>
	                    <LetterCircle size={32} letter={letter}/>
	                    <View style={styles.domain}>
	                      <Text style={{color: colors.nearBlack, fontSize: 15, fontWeight: '600'}}>{domainCapitalized}</Text>
	                      <Text style={{color: colors.gray, fontSize: 12, fontWeight: '600'}}>via Splash Chrome extension</Text>
	                    </View>
	                  </View>
	                </View>
	                <View style={styles.footer}>
	                  <Text style={styles.description}>We will load {relativeCurrency} ${relativeAmount} onto a temporary {'\n'} “magic” credit card you can use in your browser.</Text>
	                  <Button
	                  	onPress={() => approve(transaction)}
	                  	style={styles.button} 
	                  	loading={this.props.loading && !this.state.success}
	                  	checkmark={this.state.success && !this.props.loading}
	                  	checkmarkPersist={true}
											checkmarkCallback={() => dismiss()}
											disabled={this.props.error}
											title={"Approve Transaction"} primary={true}/>
	                  <View style={{flexDirection: 'row', paddingTop: 10, alignSelf: 'center', alignItems: 'center'}}>
	                    <Image style={{height: 13, width: 10}} source={require('../../assets/icons/lockIcon.png')}/>
	                    <Text style={{paddingLeft: 10, backgroundColor: 'rgba(0,0,0,0)', color: colors.lightGray, fontSize: 15, fontWeight: '600'}}>Payment secured by Splash</Text>
	                  </View>
	                </View>
								</View>
	            }

						{this.props.error &&
							<View style={styles.content}>
							<View style={styles.header}>
								<Text style={styles.title}>Transaction request</Text>
								<TouchableOpacity onPress={() => dismiss()}>
									<Image style={{height: 14, width: 14}} source={require('../../assets/icons/Xbutton.png')}/>
								</TouchableOpacity>
							</View>
							<Text style={{justifyContent: 'center', alignItems: 'center'}}>
								Oops! something went wrong when processing your
								transaction
							</Text>
							</View>}
					</View>
					</TouchableWithoutFeedback>
				</View>
			</Animated.View>
			</TouchableWithoutFeedback>

		)
	}
}

export default ApproveCardModal

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: "rgba(0,0,0,0)",
    justifyContent: 'flex-end'
	},
  popup: {
    flex: 1,
    flexDirection: "column",
    height: 474,
    alignItems: "center",
    borderTopLeftRadius: 15,
		borderTopRightRadius: 15,
    backgroundColor: colors.white
  },
  content: {
    flex: 1,
    paddingTop: 32,
    paddingBottom: 72,
    paddingHorizontal: 25,
    justifyContent: 'space-between',
    alignSelf: "stretch",
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingBottom: 10,
  },
  closeButton: {
  	height: 20,
  	width: 20,
  	margin: 8,
  },
  title: {
    fontSize: 20,
    fontWeight: "600",
    color: colors.nearBlack,
  },
  information: {
    flex: 1,
    paddingHorizontal: 32,
    justifyContent: 'space-around',
    alignItems: 'center',
  },
	exchangeText: {
		fontSize: 14,
		fontWeight: '500',
		color: colors.gray
	},
  amountBox: {
    height: 83,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'stretch',
    borderRadius: 8,
    backgroundColor: '#6364F1',
  },
  domainInfo: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 18,
  },
  domain: {
    paddingLeft: 10,
    flexDirection: 'column'
  },
  description: {
    textAlign: 'center',
    fontSize: 14,
    color: colors.gray,
    fontWeight: '500',
		paddingTop: 10,
    paddingBottom: 20,
  },
})
