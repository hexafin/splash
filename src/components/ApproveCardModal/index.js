import ApproveCardModal from "./ApproveCardModal";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { ApproveTransaction, DismissTransaction } from "../../redux/transactions/actions";

/*
Connecting ApproveCardModal to redux
*/

const mapStateToProps = state => {
	return {
		loading: state.transactions.isApprovingTransaction,
		error: state.transactions.errorApprovingTransaction
	};
};

const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{
			ApproveTransaction,
			DismissTransaction
		},
		dispatch
	);
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(ApproveCardModal);
