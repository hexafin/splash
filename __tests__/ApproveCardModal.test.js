import "react-native";
import React from "react";
import ApproveCardModal from "../src/components/ApproveCardModal/ApproveCardModal";

it("ApproveCardModal renders correctly", () => {
	const snap = shallow(
		<ApproveCardModal
			navigation={{
				state: {
					params: {
						domain: "https://github.com"
					}
				}
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});
