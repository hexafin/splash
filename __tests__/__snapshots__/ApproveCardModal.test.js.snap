// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApproveCardModal renders correctly 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0,0,0,0)",
          "flex": 1,
          "justifyContent": "flex-end",
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0)",
        },
      ]
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <TouchableWithoutFeedback
        onPress={[Function]}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderTopLeftRadius": 15,
              "borderTopRightRadius": 15,
              "flex": 1,
              "flexDirection": "column",
              "height": 474,
            }
          }
        >
          <Component
            style={
              Object {
                "alignSelf": "stretch",
                "flex": 1,
                "justifyContent": "space-between",
                "paddingBottom": 72,
                "paddingHorizontal": 25,
                "paddingTop": 32,
              }
            }
          >
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 10,
                }
              }
            >
              <Component
                style={
                  Object {
                    "color": "#333333",
                    "fontSize": 20,
                    "fontWeight": "600",
                  }
                }
              >
                Transaction request
              </Component>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Component
                  source={
                    Object {
                      "testUri": "../../../src/assets/icons/Xbutton.png",
                    }
                  }
                  style={
                    Object {
                      "height": 20,
                      "margin": 8,
                      "width": 20,
                    }
                  }
                />
              </TouchableOpacity>
            </Component>
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "space-around",
                  "paddingHorizontal": 32,
                }
              }
            >
              <Component
                style={
                  Object {
                    "color": "#919191",
                    "fontSize": 14,
                    "fontWeight": "500",
                  }
                }
              >
                1 BTC = $
                NaN
                 
              </Component>
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "#6364F1",
                    "borderRadius": 8,
                    "height": 83,
                    "justifyContent": "center",
                  }
                }
              >
                <Component
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 28,
                      "fontWeight": "600",
                    }
                  }
                >
                   $
                </Component>
                <Component
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                      "fontWeight": "600",
                      "opacity": 0.77,
                    }
                  }
                >
                  NaN
                   BTC
                </Component>
              </Component>
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingLeft": 18,
                  }
                }
              >
                <Component
                  style={
                    Object {
                      "color": "#333333",
                      "fontSize": 12,
                      "fontWeight": "700",
                      "paddingRight": 10,
                    }
                  }
                >
                  on
                </Component>
                <LetterCircle
                  letter="H"
                  size={32}
                />
                <Component
                  style={
                    Object {
                      "flexDirection": "column",
                      "paddingLeft": 10,
                    }
                  }
                >
                  <Component
                    style={
                      Object {
                        "color": "#333333",
                        "fontSize": 15,
                        "fontWeight": "600",
                      }
                    }
                  >
                    Https://github.com
                  </Component>
                  <Component
                    style={
                      Object {
                        "color": "#919191",
                        "fontSize": 12,
                        "fontWeight": "600",
                      }
                    }
                  >
                    via Splash Chrome extension
                  </Component>
                </Component>
              </Component>
            </Component>
            <Component>
              <Component
                style={
                  Object {
                    "color": "#919191",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "paddingBottom": 20,
                    "paddingTop": 10,
                    "textAlign": "center",
                  }
                }
              >
                We will load 
                 $
                 onto a temporary 
                

                 “magic” credit card you can use in your browser.
              </Component>
              <Button
                checkmark={false}
                checkmarkCallback={[Function]}
                checkmarkPersist={true}
                onPress={[Function]}
                primary={true}
                title="Approve Transaction"
              />
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "flexDirection": "row",
                    "paddingTop": 10,
                  }
                }
              >
                <Component
                  source={
                    Object {
                      "testUri": "../../../src/assets/icons/lockIcon.png",
                    }
                  }
                  style={
                    Object {
                      "height": 13,
                      "width": 10,
                    }
                  }
                />
                <Component
                  style={
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "color": "#ADADAD",
                      "fontSize": 15,
                      "fontWeight": "600",
                      "paddingLeft": 10,
                    }
                  }
                >
                  Payment secured by Splash
                </Component>
              </Component>
            </Component>
          </Component>
        </Component>
      </TouchableWithoutFeedback>
    </Component>
  </AnimatedComponent>
</TouchableWithoutFeedback>
`;
